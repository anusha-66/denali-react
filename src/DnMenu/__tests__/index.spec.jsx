/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { render } from 'enzyme';
import { DnMenu, DnMenuAlignment } from '../../index'; // src/index.tsx

describe('DnMenu Tests', () => {
    it('should export DnMenuAlignment', () => {
        expect.assertions(1);

        expect(DnMenuAlignment).toStrictEqual({
            default: '',
            left: 'left',
            right: 'right',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnMenu>
                <DnMenu.Trigger>Hover for default menu</DnMenu.Trigger>
                <DnMenu.Content>
                    <a href="#home">Home</a>
                    <a href="#about">About</a>
                    <a href="#contact">Contact</a>
                    <a href="#help">Help</a>
                    <hr />
                    <a href="#support">Support</a>
                </DnMenu.Content>
            </DnMenu>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnMenu className="Foo" alignment={DnMenuAlignment.left}>
                <DnMenu.Trigger>Hover for default menu</DnMenu.Trigger>
                <DnMenu.Content>
                    <a href="#home">Home</a>
                    <a href="#about">About</a>
                    <a href="#contact">Contact</a>
                    <a href="#help">Help</a>
                    <hr />
                    <a href="#support">Support</a>
                </DnMenu.Content>
            </DnMenu>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with isEllipses props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnMenu className="Foo" alignment={DnMenuAlignment.left}>
                <DnMenu.Trigger isEllipses>Hover for default menu</DnMenu.Trigger>
                <DnMenu.Content>
                    <a href="#home">Home</a>
                    <a href="#about">About</a>
                    <a href="#contact">Contact</a>
                    <a href="#help">Help</a>
                    <hr />
                    <a href="#support">Support</a>
                </DnMenu.Content>
            </DnMenu>,
        );

        expect(rendered).toMatchSnapshot();
    });
});
