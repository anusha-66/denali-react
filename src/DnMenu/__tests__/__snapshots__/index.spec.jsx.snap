// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnMenu Tests should render correctly with custom props 1`] = `
<div
  class="menu is-left Foo"
>
  <a
    class="menu-trigger"
  >
    Hover for default menu
  </a>
  <div
    class="menu-content"
  >
    <span
      class="link"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#about"
      >
        About
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#contact"
      >
        Contact
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#help"
      >
        Help
      </a>
    </span>
    <span
      class="link"
    >
      <hr />
    </span>
    <span
      class="link"
    >
      <a
        href="#support"
      >
        Support
      </a>
    </span>
  </div>
</div>
`;

exports[`DnMenu Tests should render correctly with isEllipses props 1`] = `
<div
  class="menu is-left Foo"
>
  <a
    class="menu-trigger"
  >
    <i
      class="d-icon d-more-horizontal is-small link"
    />
  </a>
  <div
    class="menu-content"
  >
    <span
      class="link"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#about"
      >
        About
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#contact"
      >
        Contact
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#help"
      >
        Help
      </a>
    </span>
    <span
      class="link"
    >
      <hr />
    </span>
    <span
      class="link"
    >
      <a
        href="#support"
      >
        Support
      </a>
    </span>
  </div>
</div>
`;

exports[`DnMenu Tests should render with default props 1`] = `
<div
  class="menu"
>
  <a
    class="menu-trigger"
  >
    Hover for default menu
  </a>
  <div
    class="menu-content"
  >
    <span
      class="link"
    >
      <a
        href="#home"
      >
        Home
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#about"
      >
        About
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#contact"
      >
        Contact
      </a>
    </span>
    <span
      class="link"
    >
      <a
        href="#help"
      >
        Help
      </a>
    </span>
    <span
      class="link"
    >
      <hr />
    </span>
    <span
      class="link"
    >
      <a
        href="#support"
      >
        Support
      </a>
    </span>
  </div>
</div>
`;
